pandas/tests/test_testing.py

```diff
@@ -215,6 +215,13 @@ def test_multiindex_dtype(self):
                 {'a':[1.0,2.0],'b':[2.1,1.5],'c':['l1','l2']}, index=['a','b'])
         self._assert_not_equal(df1, df2, check_index_type=True)
 
+    def test_empty_dtypes(self):
```

Comment:
I actually put this test in because it was asked for by @hayd .
It's testing whether empty dataframes with different dtypes raise an exception in `assert_frame_equal`.  I also would argue its testing *something* because this test would fail under master.
If you insist, it can be taken out , though not sure I understand the reasoning.  Why is it circular?